src/MonitorProfiler/Utilities/ThreadNameCache.cpp

```diff
@@ -0,0 +1,30 @@
+// Licensed to the.NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "ThreadNameCache.h"
+
+void ThreadNameCache::Set(ThreadID id, tstring&& name)
```

Comment:
nit: Seeing this in the rest of the code, it would not be obvious to me that the name that is passed in would be reset after this function exits. Should be rename this function to be clearer about it? I don't have a good suggestion, nor do I feel strongly about this; I just think this might trip us up later if we try to use the thread name after calling this function.